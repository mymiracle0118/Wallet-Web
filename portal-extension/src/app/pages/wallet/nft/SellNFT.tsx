import React from 'react'
import { useTranslation } from 'react-i18next'
import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import { CustomTypography } from 'app/components'

import ExternalLinkIcon from 'assets/icons/external-link.svg'

const everyone = [
  {
    name: 'OpenSea',
    websiteURL: 'https://opensea.io',
    logo: 'https://storage.googleapis.com/opensea-static/Logomark/Logomark-Blue.png',
  },
  {
    name: 'Rariable',
    websiteURL: 'https://rarible.com',
    logo: 'https://assets.coingecko.com/nft_contracts/images/1082/large/rarible.jpg',
  },
  {
    name: 'LooksRare',
    websiteURL: 'https://looksrare.com',
    logo: 'https://assets.coingecko.com/coins/images/22173/large/circle-black-256.png',
  },
]

const artist = [
  {
    name: 'SuperRare',
    websiteURL: 'https://superrare.com',
    logo: 'https://assets.coingecko.com/coins/images/17753/large/RARE.jpg',
  },
  {
    name: 'Foundation',
    websiteURL: 'https://foundation.org',
    logo: 'https://theme.zdassets.com/theme_assets/12129897/660def02924945816049919b8ec01b16dd07cf3e.svg',
  },
  {
    name: 'Nifty Gateway',
    websiteURL: 'https://www.niftygateway.com/',
    logo: 'https://media.niftygateway.com/image/upload/q_auto:good,w_500/v1576344316/nifty-builder-images/kyhclu5quebqm4sit0he.png',
  },
]

const SellNFT = () => {
  const { t } = useTranslation()

  const renderRow = ({ name, websiteURL, logo }: { name: string; websiteURL: string; logo: string }) => {
    return (
      <div key={name} className="flex items-center justify-center h-[56px]">
        <img className="h-[36px] w-[36px] rounded-full" alt="token-thumbnail" src={logo} />
        <CustomTypography className="flex-1 ml-4" variant="subtitle">
          {name}
        </CustomTypography>
        <button type="button" onClick={() => window.open(websiteURL)}>
          <ExternalLinkIcon className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]" />
        </button>
      </div>
    )
  }

  return (
    <SinglePageTitleLayout title={t('Nft.sellArtwork')}>
      <div className="px-4">
        <CustomTypography type="secondary" className="py-4" variant="subtitle">
          {t('Nft.forEveryone')}
        </CustomTypography>
        {everyone.map(({ name, websiteURL, logo }) => renderRow({ name, websiteURL, logo }))}
        <CustomTypography type="secondary" className="py-4" variant="subtitle">
          {t('Nft.forArtist')}
        </CustomTypography>
        {artist.map(({ name, websiteURL, logo }) => renderRow({ name, websiteURL, logo }))}
      </div>
    </SinglePageTitleLayout>
  )
}

export default SellNFT
