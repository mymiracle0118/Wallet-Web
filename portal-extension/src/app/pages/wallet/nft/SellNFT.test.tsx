import React from 'react'
import renderer from 'react-test-renderer'

import SellNFT from './SellNFT'

describe('<SellNFT />', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<SellNFT />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
