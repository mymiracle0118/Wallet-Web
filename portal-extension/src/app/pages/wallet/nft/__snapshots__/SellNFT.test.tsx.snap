// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellNFT /> renders correctly 1`] = `
<div
  className="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
>
  <div
    className="flex flex-row relative justify-between items-center h-16"
  >
    <button
      className="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
      onClick={[Function]}
      type="button"
    >
      <arrow-left.svg
        className="text-xl stroke-custom-black dark:stroke-custom-white"
      />
    </button>
    <div
      className="flex-1 text-center"
    >
      <div
        className="text-custom-black dark:text-custom-white text-lg font-extrabold"
        style={
          Object {
            "WebkitLineClamp": "3",
          }
        }
      >
        Nft.sellArtwork
      </div>
    </div>
  </div>
  <div
    className="w-full h-full overflow-auto"
  >
    <div
      className="MuiBox-root css-10cpdan"
    >
      <div
        className="py-4 text-custom-grey100 dark:text-custom-white40 text-sm font-extrabold"
        style={
          Object {
            "WebkitLineClamp": "3",
          }
        }
      >
        Nft.forEveryone
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://storage.googleapis.com/opensea-static/Logomark/Logomark-Blue.png"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          OpenSea
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://assets.coingecko.com/nft_contracts/images/1082/large/rarible.jpg"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          Rariable
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://assets.coingecko.com/coins/images/22173/large/circle-black-256.png"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          LooksRare
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
      <div
        className="py-4 text-custom-grey100 dark:text-custom-white40 text-sm font-extrabold"
        style={
          Object {
            "WebkitLineClamp": "3",
          }
        }
      >
        Nft.forArtist
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://assets.coingecko.com/coins/images/17753/large/RARE.jpg"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          SuperRare
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://theme.zdassets.com/theme_assets/12129897/660def02924945816049919b8ec01b16dd07cf3e.svg"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          Foundation
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
      <div
        className="flex items-center justify-center h-[56px]"
      >
        <img
          alt="token-thumbnail"
          className="h-[36px] w-[36px] rounded-full"
          src="https://media.niftygateway.com/image/upload/q_auto:good,w_500/v1576344316/nifty-builder-images/kyhclu5quebqm4sit0he.png"
        />
        <div
          className="flex-1 ml-4 text-custom-black dark:text-custom-white text-sm font-extrabold"
          style={
            Object {
              "WebkitLineClamp": "3",
            }
          }
        >
          Nifty Gateway
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          <external-link.svg
            className="fill-custom-black dark:fill-custom-white h-[24px] w-[24px]"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
